targets/arm/mikroe/sam/src/spi_master/implementations/implementation_1/hal_ll_spi_master.c

```diff
@@ -0,0 +1,661 @@
+                                        return --index_counter;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    if ( index_counter ) {
+        return hal_ll_module_id;
+    } else {
+        return HAL_LL_PIN_NC;
+    }
+}
+
+static void hal_ll_spi_master_map_pins( uint8_t module_index, hal_ll_spi_pin_id *index_list ) {
+    // if every single pin is OK, insert them into this new map, and use this map in all low level functions.
+    hal_ll_spi_master_hw_specifics_map[ module_index ].pins.sck.pin_name  = _spi_sck_map[ index_list[ module_index ].pin_sck ].pin;
+    hal_ll_spi_master_hw_specifics_map[ module_index ].pins.miso.pin_name = _spi_miso_map[ index_list[ module_index ].pin_miso ].pin;
+    hal_ll_spi_master_hw_specifics_map[ module_index ].pins.mosi.pin_name = _spi_mosi_map[ index_list[ module_index ].pin_mosi ].pin;
+
+    // SCK, MISO and MOSI pins could have different alternate function settings, hence save all the AF-s.
+    hal_ll_spi_master_hw_specifics_map[ module_index ].pins.sck.pin_af   = _spi_sck_map[ index_list[ module_index ].pin_sck ].af;
+    hal_ll_spi_master_hw_specifics_map[ module_index ].pins.miso.pin_af  = _spi_miso_map[ index_list[ module_index ].pin_miso ].af;
+    hal_ll_spi_master_hw_specifics_map[ module_index ].pins.mosi.pin_af  = _spi_mosi_map[ index_list[ module_index ].pin_mosi ].af;
+}
+
+static hal_ll_spi_master_hw_specifics_map_t *hal_ll_get_specifics( handle_t handle ) {
+    uint8_t hal_ll_module_count = sizeof( hal_ll_module_state ) / ( sizeof( hal_ll_spi_master_handle_register_t ) );
+
+    static uint8_t hal_ll_module_error = sizeof( hal_ll_module_state ) / ( sizeof( hal_ll_spi_master_handle_register_t ) );
+
+    while ( hal_ll_module_count-- ) {
+        if ( hal_ll_spi_master_get_base_from_hal_handle == hal_ll_spi_master_hw_specifics_map[ hal_ll_module_count ].base ) {
+            return &hal_ll_spi_master_hw_specifics_map[ hal_ll_module_count ];
+        }
+    }
+
+    // If NOK, return pointer to the last row of this map ( point to null pointer ).
+    return &hal_ll_spi_master_hw_specifics_map[hal_ll_module_error];
+}
+
+static void hal_ll_spi_master_set_clock( hal_ll_spi_master_hw_specifics_map_t *map, bool hal_ll_state, uint32_t *clock_value ) {
+    switch ( map->module_index ) {
+        #ifdef SPI_MODULE_0
+        case ( hal_ll_spi_master_module_num( SPI_MODULE_0 ) ):
+            ( hal_ll_state == true )?( set_reg_bit( _PMC_PCER0, HAL_LL_PID_SPI0_BIT ) ):( set_reg_bit( _PMC_PCDR0, HAL_LL_PID_SPI0_BIT ) );
+            break;
+        #endif
+        #ifdef SPI_MODULE_1
+        case ( hal_ll_spi_master_module_num( SPI_MODULE_1 ) ):
+            ( hal_ll_state == true )?( set_reg_bit( _PMC_PCER1, HAL_LL_PID_SPI1_BIT ) ):( set_reg_bit( _PMC_PCDR1, HAL_LL_PID_SPI1_BIT ) );
+            break;
+        #endif
+
```

Comment:
Add default case.

Reply:
Added it for all source files where needed.